Cubical/Core/Prelude.agda

```diff
@@ -28,102 +28,104 @@ open import Cubical.Core.Primitives public
 -- Basic theory about paths. These proofs should typically be
 -- inlined. This module also makes equational reasoning work with
 -- (non-dependent) paths.
-module _ {ℓ} {A : Set ℓ} where
-  refl : {x : A} → x ≡ x
-  refl {x} = λ _ → x
 
-  sym : {x y : A} → x ≡ y → y ≡ x
-  sym p = λ i → p (~ i)
+private
+  variable
+    ℓ ℓ' : Level
+    A : Set ℓ
+    B : A → Set ℓ'
+    x y z : A
 
-  cong : ∀ {ℓ'} {B : A → Set ℓ'} {x y : A}
-         (f : (a : A) → B a) (p : x ≡ y)
-       → PathP (λ i → B (p i)) (f x) (f y)
-  cong f p = λ i → f (p i)
+refl : x ≡ x
+refl {x = x} = λ _ → x
+
+sym : x ≡ y → y ≡ x
+sym p = λ i → p (~ i)
 
-  -- This is called compPath and not trans in order to eliminate
-  -- confusion with transp
-  compPath : {x y z : A} → x ≡ y → y ≡ z → x ≡ z
-  compPath {x = x} p q i =
-    hcomp (λ j → \ { (i = i0) → x
-                   ; (i = i1) → q j }) (p i)
+cong : ∀ (f : (a : A) → B a) (p : x ≡ y)
+       → PathP (λ i → B (p i)) (f x) (f y)
+cong f p = λ i → f (p i)
 
-  infix  3 _≡-qed _∎
-  infixr 2 _≡⟨⟩_ _≡⟨_⟩_
-  infix  1 ≡-proof_ begin_
+-- This is called compPath and not trans in order to eliminate
+-- confusion with transp
+compPath : x ≡ y → y ≡ z → x ≡ z
+compPath {x = x} p q i =
+  hcomp (λ j → \ { (i = i0) → x
+                 ; (i = i1) → q j }) (p i)
 
-  ≡-proof_ begin_ : {x y : A} → x ≡ y → x ≡ y
-  ≡-proof x≡y = x≡y
-  begin_ = ≡-proof_
+infix  3 _≡-qed _∎
+infixr 2 _≡⟨⟩_ _≡⟨_⟩_
+infix  1 ≡-proof_ begin_
 
-  _≡⟨⟩_ : (x {y} : A) → x ≡ y → x ≡ y
-  _ ≡⟨⟩ x≡y = x≡y
+≡-proof_ begin_ : x ≡ y → x ≡ y
+≡-proof x≡y = x≡y
+begin_ = ≡-proof_
 
-  _≡⟨_⟩_ : (x {y z} : A) → x ≡ y → y ≡ z → x ≡ z
-  _ ≡⟨ x≡y ⟩ y≡z = compPath x≡y y≡z
+_≡⟨⟩_ : (x : A) → x ≡ y → x ≡ y
+_ ≡⟨⟩ x≡y = x≡y
 
-  _≡-qed _∎ : (x : A) → x ≡ x
-  _ ≡-qed  = refl
-  _∎ = _≡-qed
+_≡⟨_⟩_ : (x : A) → x ≡ y → y ≡ z → x ≡ z
+_ ≡⟨ x≡y ⟩ y≡z = compPath x≡y y≡z
 
+_≡-qed _∎ : (x : A) → x ≡ x
```

Comment:
Why is x being parametrized over explicitly here? Same for some of the above definitions.

Reply:
I think I didn't change the semantics, this question should be asked to the original author...

Reply:
I just copy-pasted this code from the standard library :) 

I don't know if this is used anywhere actually... Maybe it should just be removed.

Reply:
Oh. That's because the `EqReasoning` style reasoning explicitly specifies the last step of the equation transformation, so `x : A` here.